boilerplate/app/screens/DemoPodcastListScreen.tsx

```diff
@@ -39,91 +39,96 @@ const rnrImages = [rnrImage1, rnrImage2, rnrImage3]
 
 export const DemoPodcastListScreen: FC<DemoTabScreenProps<"DemoPodcastList">> = observer(
   function DemoPodcastListScreen(_props) {
-  const { episodeStore } = useStores()
+    const { episodeStore } = useStores()
 
-  const [refreshing, setRefreshing] = React.useState(false)
-  const [isLoading, setIsLoading] = React.useState(false)
+    const [refreshing, setRefreshing] = React.useState(false)
+    const [isLoading, setIsLoading] = React.useState(false)
 
-  // initially, kick off a background refresh without the refreshing UI
-  useEffect(() => {
-    ;(async function load() {
-      setIsLoading(true)
-      await episodeStore.fetchEpisodes()
-      setIsLoading(false)
-    })()
-  }, [episodeStore])
+    // initially, kick off a background refresh without the refreshing UI
+    useEffect(() => {
+      ;(async function load() {
+        setIsLoading(true)
+        await episodeStore.fetchEpisodes()
+        setIsLoading(false)
+      })()
+    }, [episodeStore])
```

Comment:
A lot of querying libraries already do that. mst-query, react-query, urql, etc.. Since we don't use that, it might be cool to have a generic hook for any promises (something like this: https://github.com/bsonntag/react-use-promise#usage). 

Also, isn't react coming out with `use()` hook that will have something simiarl?